SimPEG/EM/Base.py

```diff
@@ -202,16 +204,26 @@ def MfMui(self):
             self._MfMui = self.mesh.getFaceInnerProduct(self.mui)
         return self._MfMui
 
-    def MfMuiDeriv(self, u):
+    def MfMuiDeriv(self, u, v=None, adjoint=False):
         """
         Derivative of :code:`MfMui` with respect to the model.
         """
         if self.muiMap is None:
             return Utils.Zero()
 
-        return (
-            self.mesh.getFaceInnerProductDeriv(self.mui)(u) * self.muiDeriv
-        )
+        if getattr(self, '_MfMuiDeriv', None) is None:
+            self._MfMuiDeriv = self.mesh.getFaceInnerProductDeriv(
+                np.ones(self.mesh.nC)
+            )(np.ones(self.mesh.nF)) * self.muiDeriv
+
+        if v is not None:
```

Comment:
in this case, if v is None, we get the same behaviour as before ant this returns a matrix

Reply:
nice. it is handy for the later use when we want to generate sensitivity matrix.